WooCommerce/Classes/Authentication/Epilogue/StorePickerViewController.swift

```diff
@@ -379,11 +420,20 @@ private extension StorePickerViewController {
         RequirementsChecker.checkMinimumWooVersion(for: siteID) { [weak self] (result, error) in
             switch result {
             case .validWCVersion:
+                self?.toggleDismissButton(enabled: true)
                 self?.updateActionButtonAndTableState(animating: false, enabled: true)
             case .invalidWCVersion:
-                self?.updateActionButtonAndTableState(animating: false, enabled: false)
-                self?.displayFancyWCRequirementAlert(siteName: siteName)
+                self?.toggleDismissButton(enabled: false)
```

Comment:
🤔 this is a switch within a switch... in a method that might be a bit too long already. Would you consider extracting each of the branches in the outer switch to a separate method? That might help with readability

Reply:
done ✔️